### PR TITLE
[spirv] Add basic lowering pipeline without tiling and distribution

### LINKED ISSUE #14348: Missing SPIR-V CodeGen pipeline support for scalar dispatches
### What happened?

With this IR:

```
hal.executable public @forward_dispatch_197 {
  hal.executable.variant public @vulkan_spirv_fb, target = <"vulkan", "vulkan-spirv-fb", {spirv.target_env = #spirv.target_env<#spirv.vce<v1.6, [Shader, Float64, Float16, Int64, Int16, Int8, StorageBuffer16BitAccess, StorageUniform16, StoragePushConstant16, StorageBuffer8BitAccess, UniformAndStorageBuffer8BitAccess, StoragePushConstant8, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative, GroupNonUniformClustered, GroupNonUniformQuad, VariablePointers, VariablePointersStorageBuffer, DotProduct, DotProductInputAll, DotProductInput4x8BitPacked, DotProductInput4x8Bit, CooperativeMatrixNV], [SPV_KHR_16bit_storage, SPV_KHR_8bit_storage, SPV_KHR_integer_dot_product, SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers, SPV_NV_cooperative_matrix]>, api=Vulkan, AMD:DiscreteGPU, #spirv.resource_limits<max_compute_shared_memory_size = 65536, max_compute_workgroup_invocations = 1024, max_compute_workgroup_size = [1024, 1024, 1024], subgroup_size = 64, min_subgroup_size = 32, max_subgroup_size = 64, cooperative_matrix_properties_nv = [#spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = i8, b_type = i8, c_type = i32, result_type = i32, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f16, result_type = f16, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f32, result_type = f32, scope = <Subgroup>>]>>}> {
    hal.executable.export public @forward_dispatch_197 ordinal(0) layout(#hal.pipeline.layout<push_constants = 0, sets = [<0, bindings = [<0, storage_buffer, ReadOnly>, <1, storage_buffer>]>]>) {
    ^bb0(%arg0: !hal.device loc("<eval_with_key>.3":773:12)):
      %c1 = arith.constant 1 : index loc("<eval_with_key>.3":773:12)
      hal.return %c1, %c1, %c1 : index, index, index loc("<eval_with_key>.3":773:12)
    } loc("<eval_with_key>.3":773:12)
    builtin.module {
      func.func @forward_dispatch_197() {
        %c0 = arith.constant 0 : index loc("<eval_with_key>.3":773:12)
        %c6364136223846793005_i64 = arith.constant 6364136223846793005 : i64 loc("<eval_with_key>.3":773:12)
        %c1442695040888963407_i64 = arith.constant 1442695040888963407 : i64 loc("<eval_with_key>.3":773:12)
        %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%c0) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<i64>> loc("<eval_with_key>.3":773:12)
        %1 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%c0) : !flow.dispatch.tensor<writeonly:tensor<i64>> loc("<eval_with_key>.3":773:12)
        %2 = flow.dispatch.tensor.load %0, offsets = [], sizes = [], strides = [] : !flow.dispatch.tensor<readonly:tensor<i64>> -> tensor<i64> loc("<eval_with_key>.3":773:12)
        %extracted = tensor.extract %2[] : tensor<i64> loc("<eval_with_key>.3":773:12)
        %3 = arith.muli %extracted, %c6364136223846793005_i64 : i64 loc("<eval_with_key>.3":773:12)
        %4 = arith.addi %3, %c1442695040888963407_i64 : i64 loc("<eval_with_key>.3":773:12)
        %inserted = tensor.insert %4 into %2[] : tensor<i64> loc("<eval_with_key>.3":773:12)
        flow.dispatch.tensor.store %inserted, %1, offsets = [], sizes = [], strides = [] : tensor<i64> -> !flow.dispatch.tensor<writeonly:tensor<i64>> loc("<eval_with_key>.3":773:12)
        return loc("<eval_with_key>.3":773:12)
      } loc("<eval_with_key>.3":773:12)
    } loc("<eval_with_key>.3":773:12)
  } loc("<eval_with_key>.3":773:12)
} loc("<eval_with_key>.3":773:12)
```

You can run it with

```

(shark.venv) PS D:\G\SHARK\apps\stable_diffusion\web\stabilityai_stable-diffusion-2-1-base> iree-compile.exe  --iree-input-type=tm_tensor -iree-hal-dump-executable-sources-to=subspan --iree-hal-target-backends=vulkan --mlir-print-debuginfo --mlir-print-op-on-diagnostic=true --iree-stream-resource-index-bits=64 --iree-vm-target-index-bits=64 -iree-vulkan-target-triple=rdna3-w7900-windows  -o core.vmfb .\subspan\module_forward_dispatch_197.mlir

```
And you will hit this error:

```

subspan\module_forward_dispatch_197.mlir:13:8: error: failed to legalize operation 'hal.interface.binding.subspan' that was explicitly marked illegal
        %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%c0) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<i64>> loc("<eval_with_key>.3":773:12)
       ^
subspan\module_forward_dispatch_197.mlir:13:8: note: see current operation: %3 = "hal.interface.binding.subspan"(%0) {alignment = 64 : index, binding = 0 : index, descriptor_flags = 1 : i32, descriptor_type = #hal.descriptor_type<storage_buffer>, operand_segment_sizes = array<i32: 1, 0>, set = 0 : index} : (index) -> !flow.dispatch.tensor<readonly:tensor<i64>> loc("subspan\\module_forward_dispatch_197.mlir":13:8)
subspan\module_forward_dispatch_197.mlir:2:2: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"vulkan", "vulkan-spirv-fb", {spirv.target_env = #spirv.target_env<#spirv.vce<v1.6, [Shader, Float64, Float16, Int64, Int16, Int8, StorageBuffer16BitAccess, StorageUniform16, StoragePushConstant16, StorageBuffer8BitAccess, UniformAndStorageBuffer8BitAccess, StoragePushConstant8, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative, GroupNonUniformClustered, GroupNonUniformQuad, VariablePointers, VariablePointersStorageBuffer, DotProduct, DotProductInputAll, DotProductInput4x8BitPacked, DotProductInput4x8Bit, CooperativeMatrixNV], [SPV_KHR_16bit_storage, SPV_KHR_8bit_storage, SPV_KHR_integer_dot_product, SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers, SPV_NV_cooperative_matrix]>, api=Vulkan, AMD:DiscreteGPU, #spirv.resource_limits<max_compute_shared_memory_size = 65536, max_compute_workgroup_invocations = 1024, max_compute_workgroup_size = [1024, 1024, 1024], subgroup_size = 64, min_subgroup_size = 32, max_subgroup_size = 64, cooperative_matrix_properties_nv = [#spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = i8, b_type = i8, c_type = i32, result_type = i32, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f16, result_type = f16, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f32, result_type = f32, scope = <Subgroup>>]>>}>
  hal.executable.variant public @vulkan_spirv_fb, target = <"vulkan", "vulkan-spirv-fb", {spirv.target_env = #spirv.target_env<#spirv.vce<v1.6, [Shader, Float64, Float16, Int64, Int16, Int8, StorageBuffer16BitAccess, StorageUniform16, StoragePushConstant16, StorageBuffer8BitAccess, UniformAndStorageBuffer8BitAccess, StoragePushConstant8, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative, GroupNonUniformClustered, GroupNonUniformQuad, VariablePointers, VariablePointersStorageBuffer, DotProduct, DotProductInputAll, DotProductInput4x8BitPacked, DotProductInput4x8Bit, CooperativeMatrixNV], [SPV_KHR_16bit_storage, SPV_KHR_8bit_storage, SPV_KHR_integer_dot_product, SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers, SPV_NV_cooperative_matrix]>, api=Vulkan, AMD:DiscreteGPU, #spirv.resource_limits<max_compute_shared_memory_size = 65536, max_compute_workgroup_invocations = 1024, max_compute_workgroup_size = [1024, 1024, 1024], subgroup_size = 64, min_subgroup_size = 32, max_subgroup_size = 64, cooperative_matrix_properties_nv = [#spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = i8, b_type = i8, c_type = i32, result_type = i32, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f16, result_type = f16, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f32, result_type = f32, scope = <Subgroup>>]>>}> {
 ^
subspan\module_forward_dispatch_197.mlir:2:2: note: see current operation:
"hal.executable.variant"() ({
  "hal.executable.export"() ({
  ^bb0(%arg0: !hal.device loc("<eval_with_key>.3":773:12)):
    %0 = "arith.constant"() <{value = 1 : index}> : () -> index loc("subspan\\module_forward_dispatch_197.mlir":5:6)
    "hal.return"(%0, %0, %0) : (index, index, index) -> () loc("subspan\\module_forward_dispatch_197.mlir":6:6)
  }) {layout = #hal.pipeline.layout<push_constants = 0, sets = [<0, bindings = [<0, storage_buffer, ReadOnly>, <1, storage_buffer>]>]>, ordinal = 0 : index, sym_name = "forward_dispatch_197"} : () -> () loc("subspan\\module_forward_dispatch_197.mlir":3:4)
  "builtin.module"() ({
    "func.func"() <{function_type = () -> (), sym_name = "forward_dispatch_197"}> ({
      %0 = "arith.constant"() <{value = 0 : index}> : () -> index loc("subspan\\module_forward_dispatch_197.mlir":10:8)
      %1 = "arith.constant"() <{value = 6364136223846793005 : i64}> : () -> i64 loc("subspan\\module_forward_dispatch_197.mlir":11:8)
      %2 = "arith.constant"() <{value = 1442695040888963407 : i64}> : () -> i64 loc("subspan\\module_forward_dispatch_197.mlir":12:8)
      %3 = "hal.interface.binding.subspan"(%0) {alignment = 64 : index, binding = 0 : index, descriptor_flags = 1 : i32, descriptor_type = #hal.descriptor_type<storage_buffer>, operand_segment_sizes = array<i32: 1, 0>, set = 0 : index} : (index) -> !flow.dispatch.tensor<readonly:tensor<i64>> loc("subspan\\module_forward_dispatch_197.mlir":13:8)
      %4 = "hal.interface.binding.subspan"(%0) {alignment = 64 : index, binding = 1 : index, descriptor_type = #hal.descriptor_type<storage_buffer>, operand_segment_sizes = array<i32: 1, 0>, set = 0 : index} : (index) -> !flow.dispatch.tensor<writeonly:tensor<i64>> loc("subspan\\module_forward_dispatch_197.mlir":14:8)
      %5 = "flow.dispatch.tensor.load"(%3) {operand_segment_sizes = array<i32: 1, 0, 0, 0, 0>, static_offsets = array<i64>, static_sizes = array<i64>, static_strides = array<i64>} : (!flow.dispatch.tensor<readonly:tensor<i64>>) -> tensor<i64> loc("subspan\\module_forward_dispatch_197.mlir":15:8)
      %6 = "tensor.extract"(%5) : (tensor<i64>) -> i64 loc("subspan\\module_forward_dispatch_197.mlir":16:8)
      %7 = "arith.muli"(%6, %1) : (i64, i64) -> i64 loc("subspan\\module_forward_dispatch_197.mlir":17:8)
      %8 = "arith.addi"(%7, %2) : (i64, i64) -> i64 loc("subspan\\module_forward_dispatch_197.mlir":18:8)
      %9 = "tensor.insert"(%8, %5) : (i64, tensor<i64>) -> tensor<i64> loc("subspan\\module_forward_dispatch_197.mlir":19:8)
      "flow.dispatch.tensor.store"(%9, %4) {operand_segment_sizes = array<i32: 1, 1, 0, 0, 0, 0>, static_offsets = array<i64>, static_sizes = array<i64>, static_strides = array<i64>} : (tensor<i64>, !flow.dispatch.tensor<writeonly:tensor<i64>>) -> () loc("subspan\\module_forward_dispatch_197.mlir":20:8)
      "func.return"() : () -> () loc("subspan\\module_forward_dispatch_197.mlir":21:8)
    }) : () -> () loc("subspan\\module_forward_dispatch_197.mlir":9:6)
  }) : () -> () loc("subspan\\module_forward_dispatch_197.mlir":8:4)
  "hal.executable.variant_end"() : () -> () loc(".\\subspan\\module_forward_dispatch_197.mlir":2:3)
}) {sym_name = "vulkan_spirv_fb", sym_visibility = "public", target = #hal.executable.target<"vulkan", "vulkan-spirv-fb", {spirv.target_env = #spirv.target_env<#spirv.vce<v1.6, [Shader, Float64, Float16, Int64, Int16, Int8, StorageBuffer16BitAccess, StorageUniform16, StoragePushConstant16, StorageBuffer8BitAccess, UniformAndStorageBuffer8BitAccess, StoragePushConstant8, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative, GroupNonUniformClustered, GroupNonUniformQuad, VariablePointers, VariablePointersStorageBuffer, DotProduct, DotProductInputAll, DotProductInput4x8BitPacked, DotProductInput4x8Bit, CooperativeMatrixNV], [SPV_KHR_16bit_storage, SPV_KHR_8bit_storage, SPV_KHR_integer_dot_product, SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers, SPV_NV_cooperative_matrix]>, api=Vulkan, AMD:DiscreteGPU, #spirv.resource_limits<max_compute_shared_memory_size = 65536, max_compute_workgroup_invocations = 1024, max_compute_workgroup_size = [1024, 1024, 1024], subgroup_size = 64, min_subgroup_size = 32, max_subgroup_size = 64, cooperative_matrix_properties_nv = [#spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = i8, b_type = i8, c_type = i32, result_type = i32, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f16, result_type = f16, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f32, result_type = f32, scope = <Subgroup>>]>>}>} : () -> () loc("subspan\\module_forward_dispatch_197.mlir":2:2)
subspan\module_forward_dispatch_197.mlir:1:0: error: failed to serialize executables
subspan\module_forward_dispatch_197.mlir:1:0: note: see current operation:
"hal.executable"() ({
  "hal.executable.variant"() ({
    "hal.executable.export"() ({
    ^bb0(%arg0: !hal.device loc("<eval_with_key>.3":773:12)):
      %0 = "arith.constant"() <{value = 1 : index}> : () -> index loc("subspan\\module_forward_dispatch_197.mlir":5:6)
      "hal.return"(%0, %0, %0) : (index, index, index) -> () loc("subspan\\module_forward_dispatch_197.mlir":6:6)
    }) {layout = #hal.pipeline.layout<push_constants = 0, sets = [<0, bindings = [<0, storage_buffer, ReadOnly>, <1, storage_buffer>]>]>, ordinal = 0 : index, sym_name = "forward_dispatch_197"} : () -> () loc("subspan\\module_forward_dispatch_197.mlir":3:4)
    "builtin.module"() ({
      "func.func"() <{function_type = () -> (), sym_name = "forward_dispatch_197"}> ({
        %0 = "arith.constant"() <{value = 0 : index}> : () -> index loc("subspan\\module_forward_dispatch_197.mlir":10:8)
        %1 = "arith.constant"() <{value = 6364136223846793005 : i64}> : () -> i64 loc("subspan\\module_forward_dispatch_197.mlir":11:8)
        %2 = "arith.constant"() <{value = 1442695040888963407 : i64}> : () -> i64 loc("subspan\\module_forward_dispatch_197.mlir":12:8)
        %3 = "hal.interface.binding.subspan"(%0) {alignment = 64 : index, binding = 0 : index, descriptor_flags = 1 : i32, descriptor_type = #hal.descriptor_type<storage_buffer>, operand_segment_sizes = array<i32: 1, 0>, set = 0 : index} : (index) -> !flow.dispatch.tensor<readonly:tensor<i64>> loc("subspan\\module_forward_dispatch_197.mlir":13:8)
        %4 = "hal.interface.binding.subspan"(%0) {alignment = 64 : index, binding = 1 : index, descriptor_type = #hal.descriptor_type<storage_buffer>, operand_segment_sizes = array<i32: 1, 0>, set = 0 : index} : (index) -> !flow.dispatch.tensor<writeonly:tensor<i64>> loc("subspan\\module_forward_dispatch_197.mlir":14:8)
        %5 = "flow.dispatch.tensor.load"(%3) {operand_segment_sizes = array<i32: 1, 0, 0, 0, 0>, static_offsets = array<i64>, static_sizes = array<i64>, static_strides = array<i64>} : (!flow.dispatch.tensor<readonly:tensor<i64>>) -> tensor<i64> loc("subspan\\module_forward_dispatch_197.mlir":15:8)
        %6 = "tensor.extract"(%5) : (tensor<i64>) -> i64 loc("subspan\\module_forward_dispatch_197.mlir":16:8)
        %7 = "arith.muli"(%6, %1) : (i64, i64) -> i64 loc("subspan\\module_forward_dispatch_197.mlir":17:8)
        %8 = "arith.addi"(%7, %2) : (i64, i64) -> i64 loc("subspan\\module_forward_dispatch_197.mlir":18:8)
        %9 = "tensor.insert"(%8, %5) : (i64, tensor<i64>) -> tensor<i64> loc("subspan\\module_forward_dispatch_197.mlir":19:8)
        "flow.dispatch.tensor.store"(%9, %4) {operand_segment_sizes = array<i32: 1, 1, 0, 0, 0, 0>, static_offsets = array<i64>, static_sizes = array<i64>, static_strides = array<i64>} : (tensor<i64>, !flow.dispatch.tensor<writeonly:tensor<i64>>) -> () loc("subspan\\module_forward_dispatch_197.mlir":20:8)
        "func.return"() : () -> () loc("subspan\\module_forward_dispatch_197.mlir":21:8)
      }) : () -> () loc("subspan\\module_forward_dispatch_197.mlir":9:6)
    }) : () -> () loc("subspan\\module_forward_dispatch_197.mlir":8:4)
    "hal.executable.variant_end"() : () -> () loc(".\\subspan\\module_forward_dispatch_197.mlir":2:3)
  }) {sym_name = "vulkan_spirv_fb", sym_visibility = "public", target = #hal.executable.target<"vulkan", "vulkan-spirv-fb", {spirv.target_env = #spirv.target_env<#spirv.vce<v1.6, [Shader, Float64, Float16, Int64, Int16, Int8, StorageBuffer16BitAccess, StorageUniform16, StoragePushConstant16, StorageBuffer8BitAccess, UniformAndStorageBuffer8BitAccess, StoragePushConstant8, GroupNonUniform, GroupNonUniformVote, GroupNonUniformArithmetic, GroupNonUniformBallot, GroupNonUniformShuffle, GroupNonUniformShuffleRelative, GroupNonUniformClustered, GroupNonUniformQuad, VariablePointers, VariablePointersStorageBuffer, DotProduct, DotProductInputAll, DotProductInput4x8BitPacked, DotProductInput4x8Bit, CooperativeMatrixNV], [SPV_KHR_16bit_storage, SPV_KHR_8bit_storage, SPV_KHR_integer_dot_product, SPV_KHR_storage_buffer_storage_class, SPV_KHR_variable_pointers, SPV_NV_cooperative_matrix]>, api=Vulkan, AMD:DiscreteGPU, #spirv.resource_limits<max_compute_shared_memory_size = 65536, max_compute_workgroup_invocations = 1024, max_compute_workgroup_size = [1024, 1024, 1024], subgroup_size = 64, min_subgroup_size = 32, max_subgroup_size = 64, cooperative_matrix_properties_nv = [#spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = i8, b_type = i8, c_type = i32, result_type = i32, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f16, result_type = f16, scope = <Subgroup>>, #spirv.coop_matrix_props<m_size = 16, n_size = 16, k_size = 16, a_type = f16, b_type = f16, c_type = f32, result_type = f32, scope = <Subgroup>>]>>}>} : () -> () loc("subspan\\module_forward_dispatch_197.mlir":2:2)
  "hal.executable_end"() : () -> () loc(".\\subspan\\module_forward_dispatch_197.mlir":1:1)
}) {sym_name = "forward_dispatch_197"} : () -> () loc("subspan\\module_forward_dispatch_197.mlir":1:0)

```

### Steps to reproduce your issue

_No response_

### What component(s) does this issue relate to?

_No response_

### Version information

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by antiagainst**
This pipeline just performs bufferization and additional loop and SPIR-V lowering. Immediately this is to support dispatch containing only scalar values.

Fixes https://github.com/openxla/iree/issues/14348

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Dialect/IREECodegenAttrs.td [hunk 1]
```diff
@@ -38,20 +38,22 @@ def LLVMGPU_WarpReduction : I32EnumAttrCase<"LLVMGPUWarpReduction", 14>;
 def LLVMGPU_PackUnPack : I32EnumAttrCase<"LLVMGPUPackUnPack", 15>;
 def LLVMGPU_MatmulTensorCoreMmaSync : I32EnumAttrCase<"LLVMGPUMatmulTensorCoreMmaSync", 16>;
 
+def SPIRV_BaseLowering
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Dialect/IREECodegenAttrs.td:41]: `SPIRVDefault` to be consistent with the CPU and GPU (and VMVX) defaults?

File: compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp [hunk 2]
```diff
@@ -1632,6 +1632,23 @@ static LogicalResult setSPIRVOpConfig(const spirv::TargetEnv &targetEnv,
 static LogicalResult setConfigForKernel(const spirv::TargetEnv &targetEnv,
                                         IREE::HAL::ExecutableExportOp exportOp,
                                         func::FuncOp funcOp) {
+  // First check whether we already have workgroup count set--it's a "contract"
+  // to indicate that we should bypass all tiling and distribution to go down
+  // just the most basic lowering flow.
+  if (Block *body = exportOp.getWorkgroupCountBody()) {
+    auto retOp = cast<IREE::HAL::ReturnOp>(body->getTerminator());
+    // For scalar dispatch cases--using just one thread of one workgroup.
+    auto isOne = [](Value value) { return matchPattern(value, m_One()); };
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp:1641]: Do we need this check? as long as the workgroup count region is specified we should just do the default lowering IMO

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Looks great! Thanks for doing this. I just have one tiny question.

File: compiler/src/iree/compiler/Codegen/Dialect/IREECodegenAttrs.td (hunk 1 above)
Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/Dialect/IREECodegenAttrs.td:41]: Default is not descriptive to me--it says nothing about what the pipeline does. Also here this pipeline is not really the default; it's actually a special case. The default one should be BaseVectorize, given that's the most commonly picked one. So I prefer to stay this name to be precise.

File: compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp (hunk 2 above)
Comment (antiagainst) [reply compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp:1641]: Just being conservative and not enable a lot of cases and see a surprise in the future--we can always drop later if other needs arise. :)

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**